src/rosdiscover/observer/ros1.py

```diff
@@ -20,148 +19,48 @@
-                              result_related_topics: Collection[str]) -> bool:
-        """Check that the non-goal related topics are in the right collections."""
-        return _Node.has_cancel(action, ros, goal_related_topics) and \
-            _Node.has_status(action, ros, result_related_topics) and \
-            _Node.has_feedback(action, fmt, ros, result_related_topics) and \
-            _Node.has_result(action, fmt, ros, result_related_topics)
-
-    @staticmethod
-    def has_cancel(action: str, ros: ROS1, topics: Collection[str]) -> bool:
-        cancel = f"{action}/cancel"
-        try:
-            return cancel in topics and ros.topic_to_type[cancel] == "actionlib_msgs/GoalID"
-        except KeyError:
-            logger.error(f"Topic {cancel} does not have a type.")
-            return False
-
-    @staticmethod
-    def has_status(action: str, ros: ROS1, topics: Collection[str]) -> bool:
-        status = f"{action}/status"
-        try:
-            return status in topics and ros.topic_to_type[
-                status] == "actionlib_msgs/GoalStatusArray"
-        except KeyError:
-            logger.error(f"Topic {status} does not have a type.")
-            return False
-
-    @staticmethod
-    def has_feedback(action: str, fmt: str, ros: ROS1, topics: Collection[str]) -> bool:
-        feedback = f"{action}/feedback"
-        try:
-            return feedback in topics and ros.topic_to_type[feedback] == f"{fmt}Feedback"
-        except KeyError:
-            logger.error(f"Topic {feedback} does not have a type.")
-            return False
-
-    @staticmethod
-    def has_result(action: str, fmt: str, ros: ROS1, topics: Collection[str]) -> bool:
-        result = f"{action}/result"
-        try:
-            return result in topics and ros.topic_to_type[result] == f"{fmt}Result"
-        except KeyError:
-            logger.error(f"Topic {result} does not have a type.")
-            return False
-
 
 class ROS1Observer(Observer):
 
-    def __init__(self, app: AppInstance, config: 'Config'):
+    def __init__(self, app: AppInstance, config: 'Config') -> None:
         super().__init__(app, config)
 
-    def observe_and_summarise(self) -> SystemSummary:
+    def observe(self) -> SystemSummary:
+        """Observe the state of the running system and produce a summery of the archtiecture."""
+        nodecontexts: Set[NodeContext] = set()
         with self._app_instance.ros1() as ros:
             info = ros.state
             nodes = self._transform_info(info)
             for node in nodes:
                 nodecontext = self._make_node_context(node, ros)
-                self._nodes[nodecontext.name] = nodecontext
-            return self.summarise()
+                nodecontexts.add(nodecontext)
+
+        return self._summarise(nodecontexts)
+
+    def _summarise(self, contexts: Collection[NodeContext]) -> SystemSummary:
+        """Produces an immutable description of the system architecture.
```

Comment:
For the sake of clarity, it might be worth updating this description to emphasise that the description is based on provided node contexts.

Reply:
Done